packages/core/src/lib/stream_manager.ts

```diff
@@ -0,0 +1,68 @@
+import type { PeerUpdate } from "@libp2p/interface";
+import type { Stream } from "@libp2p/interface/connection";
+import { Peer } from "@libp2p/interface/peer-store";
+import { Libp2p } from "@waku/interfaces";
+import { selectConnection } from "@waku/utils/libp2p";
+import debug from "debug";
+
+export class StreamManager {
+  private streamPool: Map<string, Promise<Stream>>;
+
+  constructor(
+    public multicodec: string,
+    public getConnections: Libp2p["getConnections"],
+    public addEventListener: Libp2p["addEventListener"],
+    private log: debug.Debugger
```

Comment:
I think we can drop this dependency and just use local `logger` with `multi codec`

Reply:
yeah, what I mean is we don't need to pass `log` always and better to define another logger for `StreamManager` like 
```ts
const log = debug("waku:stream-manager");
...
this.log(`${this.multicodec}\tOptimistically opening a stream to ${peer.id.toString()}`);
```
that would reduce need to pass another parameter to constructor